champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/assassin/bow/SmokeArrow.java

```diff
@@ -0,0 +1,106 @@
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import me.mykindos.betterpvp.champions.Champions;
+import me.mykindos.betterpvp.champions.champions.ChampionsManager;
+import me.mykindos.betterpvp.champions.champions.skills.data.SkillActions;
+import me.mykindos.betterpvp.champions.champions.skills.types.PrepareArrowSkill;
+import me.mykindos.betterpvp.core.components.champions.Role;
+import me.mykindos.betterpvp.core.components.champions.SkillType;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilMessage;
+import org.bukkit.Location;
+import org.bukkit.Particle;
+import org.bukkit.Sound;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.block.Action;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+@Singleton
+@BPvPListener
+public class SmokeArrow extends PrepareArrowSkill {
+
+    private double baseDuration;
+
+    @Inject
+    public SmokeArrow(Champions champions, ChampionsManager championsManager) {
+        super(champions, championsManager);
+    }
+
+    @Override
+    public String getName() {
+        return "Smoke Arrow";
+    }
+
+    @Override
+    public String[] getDescription(int level) {
+        return new String[] {
+                "Your next arrow will give Blindness",
+                "and Slow 2 to the target for <val>" + getEffectDuration(level) + "</val> seconds.",
+                "",
+                "Cooldown: <val>" + getCooldown(level)
+        };
+    }
+
+    private double getEffectDuration(int level) {
+        return baseDuration + level;
+    }
+
+    @Override
+    public Role getClassType() {
+        return Role.ASSASSIN;
+    }
+
+    @Override
+    public SkillType getType() {
+        return SkillType.BOW;
+    }
+
+    @Override
+    public double getCooldown(int level) {
+        return cooldown - (2 * level);
+    }
+
+    @Override
+    public void activate(Player player, int level) {
+        if (!active.contains(player.getUniqueId())) {
+            player.getWorld().playSound(player.getLocation(), Sound.ENTITY_BLAZE_AMBIENT, 2.5F, 2.0F);
+            active.add(player.getUniqueId());
+        }
+    }
+
+    @Override
+    public Action[] getActions() {
+        return SkillActions.LEFT_CLICK;
+    }
+
+    @Override
+    public void onHit(Player damager, LivingEntity target, int level) {
+        final int effectDuration = (int) (getEffectDuration(level) * 20L);
+        target.addPotionEffect(new PotionEffect(PotionEffectType.BLINDNESS, effectDuration, 1));
+        target.addPotionEffect(new PotionEffect(PotionEffectType.SLOW, effectDuration, 1));
+        target.getWorld().playSound(target.getLocation(), Sound.ENTITY_BLAZE_AMBIENT, 2.5F, 2.0F);
+
+        UtilMessage.simpleMessage(target, getClassType().getName(), "<alt2>%s</alt2> hit you with <alt>%s</alt>.", damager.getName(), getName());
```

Comment:
what if the target is not a player

Reply:
It shouldn't break anything, LivingEntity objects are still CommandSenders which can receive messages / potion effects